docs/container-quickstart.rst

```diff
@@ -0,0 +1,931 @@
+port 80. The ``--http-socket=ADDR:PORT`` will override this behavior.
+Other options given to ``docker run`` will also be passed to the
+``uwsgi`` command as well.
+
+The ``sosgui`` WSGI application requires two configuration files:
+
+#. ``dsosd.conf``: containing a list of hostnames of dsosd, one per
+   line. See `here <https://github.com/ovis-hpc/ldms-containers/blob/master/test/test-maestro/files/dsosd.conf>`__ for an
+   example.
+#. ``settings.py``: containing a WSGI application settings. Please pay
+   attention to DSOS_ROOT and DSOS_CONF. See
+   `here <https://github.com/ovis-hpc/ldms-containers/blob/master/test/test-maestro/files/settings.py>`__ for an example.
+
+Usage example:
+
+.. code:: sh
+
+   ## On a service node
+
+   # Pull the container image
+   $ docker pull ovishpc/ldms-ui
+
+   # Start ldms-ui container, using host network namespace
+   $ docker run -d --name=ui --network=host --privileged \
+	   -v /HOST/dsosd.conf:/opt/ovis/etc/dsosd.conf \
+	   -v /HOST/settings.py:/opt/ovis/ui/sosgui/settings.py \
+            ovishpc/ldms-ui
+
+LDMS-Grafana Container
+----------------------
+
+.. code:: sh
+
+   # SYNOPSIS
+   $ docker run -d --name=<CONTAINER_NAME> --network=host --privileged
+            [ -v /on-host/grafana.ini:/etc/grafana/grafana.ini:ro ] # custom grafana config
+            [ -e GF_SERVER_HTTP_ADDR=<ADDR> ] # env var to override Grafana IP address binding (default: all addresses)
+            [ -e GF_SERVER_HTTP_PORT=<PORT> ] # env var to override Grafana port binding (default: 3000)
+            ovishpc/ldms-grafana # the image name
+                 [ OTHER GRAFANA-SERVER OPTIONS ] # other options to grafana-server
+
+``ovishpc/ldms-grafana`` is based on
+`grafana/grafana-oss:9.1.0-ubuntu <https://hub.docker.com/layers/grafana/grafana/grafana/9.1.0-ubuntu/images/sha256-39ea2186a2a5f04d808342400fe667678fd02632e62f2c36efa58c27a435d31d?context=explore>`__
+with Sos data source plugin to access distributed-SOS data. The grafana
+server listens to port 3000 by default. The options specified at the
+``docker run`` CLI will be passed to the ``grafana-server`` command.
+
+.. code:: sh
+
+   ## On a service node
+
+   # Pull the container image
+   $ docker pull ovishpc/ldms-grafana
+
+   # Start ldms-grafana container, this will use port 3000
+   $ docker run -d --name=grafana --privileged --network=host ovishpc/ldms-grafana
+
+   # Use a web browser to navigate to http://HOSTNAME:3000 to access grafana
+
+SSH port forwarding to grafana
+------------------------------
+
+In the case that the grafana server cannot be accessed directly, use SSH
+port forwarding as follows:
+
+.. code:: sh
+
+   (laptop) $ ssh -L 127.0.0.1:3000:127.0.0.1:3000 LOGIN_NODE
+   (LOGIN_HODE) $ ssh -L 127.0.0.1:3000:127.0.0.1:3000 G_HOST
+   # Assuming that the ldms-grafana container is running on G_HOST.
+
+Then, you should be able to access the grafana web server via
+``http://127.0.0.1:3000/`` on your laptop.
+
+Building Containers
```

Comment:
Maybe this documentation belongs in the ovis-hpc/ldms-containers project rather than here in the ovis-hpc/ldms project?

Reply:
Oh, I was not aware of the ldms-containers subproject. This .rst file is the same as the README.md for this repo so I will go ahead and remove it.